### PR TITLE
fix(sample-gen): Generate the correct number of additionalProperties when minProperties is used

### PR DESCRIPTION
**Pull request by hkosova**
### Description

This PR fixes an issue where an empty example was generated for an object with `additionalProperties` and `minProperties: 1`. And in general, examples for `minProperties: N` + `additionalProperties` were generated with N-1 properties.

The issue occurred because the loop that added additionalProps used `<` instead of `<=` to check the exit condition. As a result, it ran one less iteration than needed - for example, it never executed when `toGenerateCount` was 1.
https://github.com/swagger-api/swagger-ui/blob/de3d063ca3c98ffeb761dbe3fee847862ca33f50/src/core/plugins/samples/fn.js#L484

### Motivation and Context
Fix/improve example generation.

### How Has This Been Tested?
Tested manually + added tests.

#### Example 1
Sample schema:
```yaml
type: object
minProperties: 1
additionalProperties:
  type: string
```

Result before the fix:
```json
{}
```

After the fix:
```json
{
  "additionalProp1": "string"
}
```

#### Example 2
Sample schema:
```yaml
type: object
minProperties: 2
additionalProperties:
  type: string
```

Result before the fix:
```json
{
  "additionalProp1": "string"
}
```

After the fix:
```json
{
  "additionalProp1": "string",
  "additionalProp2": "string"
}
```

### Screenshots (if appropriate):
n/a

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by tim-lai — CHANGES_REQUESTED** [review]
More conventionally, the for loop should be written as `for (let i = 0; i < toGenerateCount)`

I'm not sure where the default "4" comes from (vs. "3").

File: src/core/plugins/samples/fn.js [hunk 1]
```diff
@@ -480,17 +480,17 @@ export const sampleFromSchemaGeneric = (schema, config={}, exampleOverride = und
       } else {
         const toGenerateCount = schema.minProperties !== null && schema.minProperties !== undefined && propertyAddedCounter < schema.minProperties
           ? schema.minProperties - propertyAddedCounter
-          : 4
-        for (let i = 1; i < toGenerateCount; i++) {
+          : 3
+        for (let i = 0; i < toGenerateCount; i++) {
```
Comment (char0n) [inline src/core/plugins/samples/fn.js:484]: @hkosova,

Wouldn't the following loop definition solve the same thing without need to change other code?

```js
for (let i = 1; i <= toGenerateCount; i++) {
``

Comment (hkosova) [reply src/core/plugins/samples/fn.js:484]: @char0n yes, and this was in the [original commit](https://github.com/swagger-api/swagger-ui/pull/7432/commits/0f94540ec5aefbe1307037f4f3e64ebdfedd4203#diff-879e3fb9c89a199d185458ea296ccf500d8589642c726e5d0e287dcbca7a0271), but @tim-lai previously asked to use a 0-based loop:

> More conventionally, the for loop should be written as `for (let i = 0; i < toGenerateCount)`

I can change back if needed - let me know.

<br/>

Note that the `"additionalPropNN"` names should remain 1-based to match the names generated by Swagger Inflector.

Comment (char0n) [reply src/core/plugins/samples/fn.js:484]: @hkosova yes please, revert to original commit. The less code we modify the better the PR is.

**Review by char0n — APPROVED** [review]
Right, looks good to go. Bugfix consisting of two symbols changes is almost a record! ;]

**Pull request merged**